File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    # 3, 4, 5, 6 - 7
    # 
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        size = len(nums)
        storage = {}

        for i in range(size): 
            val = nums[i]
            req = target - nums[i]
            
            if req in storage.keys():
                return [storage[req], i]
            
            storage[val] = i

        return []

File: Data Structures & Algorithms/sign-of-the-product-of-an-array/submission-3.java
class Solution {
    public int arraySign(int[] nums) {
        
        long p = 1;

        for (int n : nums) {
            int s = signFunc(p * n);
            p = s;
        }

        return signFunc(p);
    }

    public int signFunc(long x) {
        if (x > 0) return 1;
        else if (x < 0) return -1;
        else return 0;
    }
}